cluster-autoscaler/cloudprovider/cloud_provider.go

```diff
@@ -154,6 +154,14 @@ type CloudProvider interface {
 	// Refresh is called before every main loop and can be used to dynamically update cloud provider state.
 	// In particular the list of node groups returned by NodeGroups can change as a result of CloudProvider.Refresh().
 	Refresh() error
+
+	// IsNodeCandidateForDeletion returns whether the node is a good candidate for deletion. This function
+	// will be called during prefiltering of nodes for scaledown to allow cloud providers the opportunity
+	// to reject a node for deletion. This may be used in cases where nodes are undergoing upgrades or other
+	// cloud-specific behavior where the cluster autoscaler should not begin cordoning, draining, and tainting
+	// the node.
+	// Returns true if the node can be safely deleted or false otherwise.
+	IsNodeCandidateForDeletion(*apiv1.Node) (bool, error)
```

Comment:
```suggestion
	IsNodeCandidateForScaleDown(*apiv1.Node) (bool, error)
```
Wondering if this should be named `IsNodeCandidateForScaleDown` instead